plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -112,15 +115,18 @@
     private final TupleDomain<IcebergColumnHandle> dataColumnPredicate;
     private final Domain pathDomain;
     private final Domain fileModifiedTimeDomain;
+    private final OptionalLong limit;
 
     private TupleDomain<IcebergColumnHandle> pushedDownDynamicFilterPredicate;
     private CloseableIterable<FileScanTask> fileScanIterable;
     private long targetSplitSize;
     private CloseableIterator<FileScanTask> fileScanIterator;
-    private Iterator<FileScanTask> fileTasksIterator;
+    private Iterator<FileScanTask> fileTasksIterator = emptyIterator();
+    private boolean fileHasAnyDeletions;
 
     private final boolean recordScannedFiles;
     private final ImmutableSet.Builder<DataFileWithDeleteFiles> scannedFiles = ImmutableSet.builder();
+    private long outputRowsLowerBound;
```

Comment:
I'd usually put this with the other non-final fields

Reply:
i agree it's subjective
i put it after scannedFiles because both track state changes over time
i put it separate from `fileHasAnyDeletions`, `fileTasksIterator` is they track mutable state for per-file basis

with this in mind, what do you think would be the best placement?

Reply:
Sounds reasonable to me